Add missing info population in the activity info. (#936)

luowenking1 opened the pull request.

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber-go/cadence-client?pullRequest=937) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber-go/cadence-client?pullRequest=937) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber-go/cadence-client?pullRequest=937) it.</sub>

venkat1109 commented on internal/internal_event_handlers.go:
```diff
@@ -494,11 +494,12 @@ func (wc *workflowEnvironmentImpl) ExecuteLocalActivity(params executeLocalActiv
+		callback:     callback,
+		retryPolicy:  params.RetryPolicy,
+		attempt:      params.Attempt,
+		workflowTask: &workflowTask{},
```
why is this needed ? localActivityTask shouldn't and doesn't need a non-nil workflowTask

venkat1109 commented on internal/internal_task_pollers.go:
```diff
@@ -464,15 +464,44 @@ func (lath *localActivityTaskHandler) executeLocalActivityTask(task *localActivi
+
+	if task.workflowTask != nil && task.workflowTask.task != nil {
+		taskToken = task.workflowTask.task.TaskToken
+	}
```
this doesn't make sense to me. For local activity task, these things don't really exist or mean anything

venkat1109 commented on internal/internal_task_pollers.go:
```diff
@@ -465,6 +465,9 @@ func (lath *localActivityTaskHandler) executeLocalActivityTask(task *localActivi
 	}
 
 	ctx := context.WithValue(rootCtx, activityEnvContextKey, &activityEnvironment{
+		workflowType:      &task.params.WorkflowInfo.WorkflowType,
```
just to be defensive, consider making a copy in a local var and then pass the addr of that local var here

venkat1109 commented on internal/internal_workflow_testsuite.go:
```diff
@@ -534,6 +534,38 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 	}
 }
 
+func (env *testWorkflowEnvironmentImpl) executeLocalActivityParams(
```
what is the difference between this and executeLocalActivity ? Why do you need to return concrete type here ?Given that this is unit test, can you type case inside of your test inside of creating this additional method which does the same thing ?

luowenking1 commented:
> Can you please add a test here for local activity :
> https://github.com/uber-go/cadence-client/blob/master/test/integration_test.go
> 
> Inside of your local activity function, assert that the context has the value for the workflowType

I have added the checking there but I did not create a new method. Instead I added the checking in an existence of a local activity testing.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/uber-go/cadence-client?pullRequest=937) <br/>All committers have signed the CLA.
